### ISSUE TITLE
Get panic from `leptos_axum` when try to update to 0.7

### ISSUE THREAD

**Issue by rafiyq** [leptos-rs/leptos#3094]
**Describe the bug**
I have a project using cloudflare worker. It builds without any error, but then i get panic from `leptos_axum` on the server side: `Static routes are not currently supported on WASM32 server targets.` 

**Leptos Dependencies**

Please copy and paste the Leptos dependencies and features from your `Cargo.toml`.

For example:
```toml
[dependencies]
axum  = { version = "0.7", default-features = false, optional = true }
chrono = "0.4.38"
console_error_panic_hook = { version = "0.1" }
console_log = { version = "1.0", optional = true }
getrandom = { version = "0.2.15", features = ["js"] }
gloo-net = { version = "0.6", features = ["http"] }
gloo-timers = { version = "0.3", features = ["futures"] }
leptos = "0.7.0-gamma2"
leptos_axum = { version = "0.7.0-gamma2", default-features = false, features = ["wasm"], optional = true }
leptos_meta = "0.7.0-gamma2"
leptos_router = "0.7.0-gamma2"
log = "0.4"
once_cell = "1.19"
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"
tower = { version = "0.5", optional = true }
tower-service = "0.3"
wasm-bindgen = { version = "0.2.92", optional = true }
wasm-bindgen-futures = "0.4"
worker = { version="0.4", features=['http', 'axum', 'd1'], optional = true }

[features]
hydrate = ["leptos/hydrate"]
ssr = [
  "dep:axum",
  "leptos/ssr",
  "leptos_axum/wasm",
  "leptos_meta/ssr",
  "leptos_router/ssr",
  "dep:leptos_axum",
  "dep:worker",
  "dep:console_log",
]
```

**To Reproduce**
1. create project using `cargo generate cloudflare/workers-rs`, select leptos
2. update leptos dependency to 0.7 on Cargo.toml
3. build and run the project using `wrangler dev`

**Expected behavior**
There aren't supposed to have any error.

**Additional context**
The problem arised when i try to update to 0.7, in 0.6 it's fine.

**Comment by benwis** [comment]
I'm also monitoring an issue by the maintainer for this specific thing. I think I need to gate the static route stuff behind a feature or figure out why it isn't being excluded

**Comment by gbj** [comment]
Should be fixed by #3097 but let me know if not.
